lib/std/js/jssynchttpclient.nim

```diff
@@ -0,0 +1,60 @@
+## Synchronous `HttpClient <httpclient.html>`_ for JavaScript implemented on top of `jsxmlhttprequest <jsxmlhttprequest.html>`_
+when not defined(js):
+  {.fatal: "Module jssynchttpclient is designed to be used with the JavaScript backend.".}
+
+import std/uri, std/js/jsxmlhttprequest
+from std/uri import Uri
+
+type JsHttpClient* = ref object of XMLHttpRequest
+
+func newJsHttpClient*(): JsHttpClient = discard
+
+proc xmlHttpRequestImpl(self: JsHttpClient; url: Uri | string; body: string; `method`: static[cstring]): cstring =
+  self.open(`method` = `method`, url = cstring($url), false)
+  self.send(body = body.cstring)
+  self.responseText
+
+proc getContent*(self: JsHttpClient; url: Uri | string): cstring =
```

Comment:
Given that this API is completely new, should we only allow `Uri` and not `string` ?.